src/Interpreters/Context.cpp

```diff
@@ -2821,6 +2821,10 @@ StorageID Context::resolveStorageIDImpl(StorageID storage_id, StorageNamespace w
     }
 
     bool look_for_external_table = where & StorageNamespace::ResolveExternal;
+    /// Global context should not contain temporary tables
+    if (isGlobalContext() && getApplicationType() != ApplicationType::LOCAL)
```

Comment:
Btw, the second part of the condition (`getApplicationType() != ApplicationType::LOCAL`) is probably outdated. AFAIK, after #26231 clickhouse-local executes queries in separate query context, not in global one.

Reply:
Indeed, nice catch! Fixed.